tests/subsys/bluetooth/mesh/scheduler_model/message_validity/src/main.c

```diff
@@ -0,0 +1,477 @@
+		BT_MESH_SCHEDULER_MSG_LEN_ACTION_SET);
+
+	scheduler_action_pack(&buf, 0, test_entry);
+
+	return _bt_mesh_scheduler_setup_srv_op[ack ? 0 : 1].func(sched_mod, NULL, &buf);
+}
+
+/** Call server GET opcode handler */
+static int current_register_get(struct bt_mesh_model *sched_mod)
+{
+	BT_MESH_MODEL_BUF_DEFINE(buf, BT_MESH_SCHEDULER_OP_GET, BT_MESH_SCHEDULER_MSG_LEN_GET);
+
+	return _bt_mesh_scheduler_srv_op[0].func(sched_mod, NULL, &buf);
+}
+
+/** Call server ACTION GET opcode handler */
+static int action_get(struct bt_mesh_model *sched_mod, uint8_t test_idx)
+{
+	BT_MESH_MODEL_BUF_DEFINE(buf, BT_MESH_SCHEDULER_OP_ACTION_GET,
+				 BT_MESH_SCHEDULER_MSG_LEN_ACTION_GET);
+
+	net_buf_simple_add_u8(&buf, test_idx);
+
+	return _bt_mesh_scheduler_srv_op[1].func(sched_mod, NULL, &buf);
+}
+
+/** Send an acked and an unacked SET message with scheduler entry 'test_entry',
+ * schedule action index count 'idx' and assert return value 'ret_val' for both.
+*/
+static void cli_set_msg_send_assert(struct bt_mesh_schedule_entry *test_entry, uint8_t test_idx,
+				    int ret_val)
+{
+	zassert_equal(bt_mesh_scheduler_cli_action_set(
+			      &sched_cli, NULL, test_idx,
+			      (const struct bt_mesh_schedule_entry *)test_entry, NULL),
+		      ret_val);
+	zassert_equal(bt_mesh_scheduler_cli_action_set_unack(
+			      &sched_cli, NULL, test_idx,
+			      (const struct bt_mesh_schedule_entry *)test_entry),
+		      ret_val);
+}
+
+static void *setup(void)
+{
+	mod_elem_idx_prep();
+
+	/* 1st of Jan 2010 */
+	start_tm.tm_year = 110;
+	start_tm.tm_mon = 0;
+	start_tm.tm_mday = 1;
+	start_tm.tm_wday = 5;
+	start_tm.tm_hour = 0;
+	start_tm.tm_min = 0;
+	start_tm.tm_sec = 0;
+
+	/* all valid action values */
+	valid_entry.year = 0x63;
+	valid_entry.month = ANY_MONTH;
+	valid_entry.day = 5;
+	valid_entry.hour = 5;
+	valid_entry.minute = 5;
+	valid_entry.second = 30;
+	valid_entry.day_of_week = ANY_DAY_OF_WEEK;
+	valid_entry.action = BT_MESH_SCHEDULER_TURN_ON;
+	valid_entry.transition_time = model_transition_encode(2500);
+	valid_entry.scene_number = 1;
+
+	zassert_not_null(_bt_mesh_scheduler_srv_cb.init, "Server init cb is null");
+	zassert_not_null(_bt_mesh_scheduler_cli_cb.init, "Client init cb is null");
+
+	_bt_mesh_scheduler_srv_cb.init(sched_mod1_elem);
+	_bt_mesh_scheduler_cli_cb.init(sched_mod2_elem);
+
+	return NULL;
+}
+
+static void run_before(void *f)
+{
+	k_sem_reset(&action_set_cb_called);
+	k_sem_reset(&action_status_rcvd);
+	k_sem_reset(&status_rcvd);
+	k_sem_reset(&status_send);
+}
+
+static void teardown(void *f)
+{
+	zassert_not_null(_bt_mesh_scheduler_srv_cb.reset, "Server reset cb is null");
+	zassert_not_null(_bt_mesh_scheduler_cli_cb.reset, "Client reset cb is null");
+
+	_bt_mesh_scheduler_srv_cb.reset(sched_mod1_elem);
+	_bt_mesh_scheduler_cli_cb.reset(sched_mod2_elem);
+}
+
+ZTEST_SUITE(scheduler_message_validity_test, NULL, setup, run_before, NULL, teardown);
+
+ZTEST(scheduler_message_validity_test, test_cli_set_prohibited)
```

Comment:
Only checked for client, not for server? Is there a reason for this?

Reply:
What, there is no prohibited value for Day, is there? All 32 values are legal. I'm talking about testing how the server handles being sent one of the values listed as Prohibited in the spec, such as hour > 0x19 (field (of 5 bits) can fit up to 0x1f), etc.

Reply:
Right, that is a client-side only issue. If the field bits are 0b00001 the server can't be expected to parse it as anything but day = 1